src/wasm/wasm-validator.cpp

```diff
@@ -3500,6 +3501,32 @@ static void validateFeatures(Module& module, ValidationInfo& info) {
   }
 }
 
+static void validateClosedWorldInterface(Module& module, ValidationInfo& info) {
+  // Error if there are any publicly exposed heap types beyond the types of
+  // publicly exposed functions.
```

Comment:
Do we explain why this is anywhere? Might be worth adding a comment on the reasons we decided to define things that way, either here or in `pass.h`.

Reply:
Would it make sense to explain this in the `--help` text for `--closed-world`?

Reply:
I think the `--help` text is usually pretty brief. We've been leaving longer details to places like `pass.h` for trapsNeverHappen etc. Though that isn't a great place, really. Maybe the main markdown page is better, which has a section on the IR? Anyhow, for now I'd say let's either add to `pass.h` as before, or to the markdown.